src/components/ProductList.jsx

```diff
@@ -0,0 +1,174 @@
+        limit: pageSize,
+        sort: sortType,
+        search: searchContent,
+      });
+
+      let products = response.list || [];
+
+      // 검색 필터링 -> 서버에서 검색 지원이 안됨
+      if (searchContent) {
+        products = products.filter((product) =>
+          product.name.toLowerCase().includes(searchContent.toLowerCase())
+        );
+      }
+      // 정렬
+      if (sortType === "recent") {
+        products.sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt));
+      } else if (sortType === "favorite") {
+        products.sort((a, b) => b.favoriteCount - a.favoriteCount);
+      }
+
+      setProducts(products);
+      setTotalPages(Math.ceil(response.totalCount / pageSize));
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  /* 핸들 */
+
+  // 정렬 변경 처리
+  const handleSortChange = (e) => {
+    setSortType(e.target.value);
+    setCurrentPage(1); // 정렬 변경 시 페이지 초기화
+  };
+
+  // 검색 처리
+  const handleSearchChange = (e) => {
+    setSearchContent(e.target.value);
+    setCurrentPage(1); // 검색 시 페이지 초기화
+  };
+
+  //이미지가 없을 때
+  const handleImageError = (e) => {
+    e.target.src = nonImage; // 대체 이미지 경로
+  };
+
+  // 페이지네이션 처리
+  const handlePrevious = () => {
+    if (currentPage > 1) setCurrentPage(currentPage - 1);
+  };
+
+  const handleNext = () => {
+    if (currentPage < totalPages) setCurrentPage(currentPage + 1);
+  };
+
+  return (
+    <div className="product-wrapper">
+      {/* 판매중 상품 */}
+      <div className="on-sale-product">
+        <div className="search-sort-bar">
+          <h2>판매 중인 상품</h2>
+          <div className="form-wrapper">
+            <input
+              className="search-input"
+              type="text"
+              placeholder="&nbsp; &nbsp; &nbsp;검색할 상품을 입력해주세요"
+              value={searchContent}
+              onChange={handleSearchChange}
+            />
+            <button className="register-product">상품 등록하기</button>
+            <select value={sortType} onChange={handleSortChange}>
+              <option value="recent">최신순</option>
+              <option value="favorite">좋아요순</option>
+            </select>
+          </div>
+        </div>
+        <div className="product-list">
+          {products.length === 0 ? (
+            <p>검색된 상품이 없습니다.</p> // 검색 결과가 없는 경우
+          ) : (
+            products.map((product) => (
+              <div key={product.id} className="product-item">
+                <img
+                  src={product.images[0] || nonImage}
+                  onError={handleImageError}
+                  alt={product.name}
+                />
+                <div className="product-content">
+                  <h3>{product.name}</h3>
+                  <p>{product.price.toLocaleString()}원</p>
+                  <div className="like-wrapper">
+                    <img src={likeButton} alt="좋아요 버튼" />
+                    <p>{product.favoriteCount}</p>
+                  </div>
+                </div>
+              </div>
+            ))
```

Comment:
@gahyeon-yang 이런거는 child component 로 사용하셔도 좋을거같아요! 
```suggestion
            products.map((product) => (
              <ProductItem product={product}/>
            ))
```